tests/Properties.hs

```diff
@@ -1614,6 +1614,36 @@ prop_read_write_file_D x = unsafePerformIO $ do
         (const $ do y <- D.readFile f
                     return (x==y))
 
+prop_hgetline_like_s8_hgetline (LinedASCII filetext) (lineEndIn, lineEndOut) = idempotentIOProperty $ do
+    tid <- myThreadId
+    let f = "qc-test-"++show tid
+    let newlineMode = NewlineMode (if lineEndIn then LF else CRLF) (if lineEndOut then LF else CRLF)
```

Comment:
It's a pity that `QuickCheck` does not provide `Arbitrary` instances for `Newline` and `NewlineMode`.

Reply:
I raised an issue (nick8325/quickcheck#322) for this, so hopefully `QuickCheck` will provide those instances soon.
But I don't know what the timeline would look like to get rid of the `Bool`s here.

Reply:
Yeah, it is not worth for `bytestring` to depend on the very latest `QuickCheck` only for the sake of these instances, so we should probably keep `if lineEndIn then LF else CRLF`. But it would be nice to submit a PR to `QuickCheck` adding them, for generations to come.

Reply:
[Pull request made](https://github.com/nick8325/quickcheck/pull/323), in the process I realized I should flip the conditional because QuickCheck attempts to shrink `True` into `False` and I would prefer shrinking `CRLF` into `LF` given that `CRLF` is the more complicated case.